### DISCUSSION TITLE
Using KG or EI and SGD alternatingly

### DISCUSSION THREAD

**Discussion by sambitmishra98** [meta-pytorch/botorch#1390] [Ideas]
I have been using Knowledge Gradient method and Expected improvement acquisition functions for my work. Based on the last few months of fiddling around with it (purely application based work), I have come to realise that the best way to go about using Bayesian optimisation to arrive at feasible or even good candidates is by first exploring the domain using `qKG` acquisition function with `SingleTaskGP `model and then `qNEI` acquisition function with `HeteroskedasticSingleTaskGP` model. 

Now after generating the dataset using the above methods, with a reasonable model at hand, we can safely assume that the domain may have multiple local minimas, with some datapoints in the near viscinity of these local minimas. A probabilistic model with information on gradients should be enough to let us use SGD.

If the cost function is low enough and SGD can be used with one of the datapoints as start point, we could get to the local minima that the datapoint would fall into. I was thinking if SGD method could be used this way to replace each candidate from the dataset with a corresponding candidate which is at the nearest local minima. Since adding all the points would increase the number of datapoints by quiet a lot, appending all the costs calculated from SGD would probably not be a good idea.

Let me know if this is a direction worth thinking about and implementing.

**Comment by Balandat** [comment]
Hmm I am not sure I understand what the goal of this approach is or whether I understand exactly what you're suggesting. It seems that you're trying to combine Bayesian optimization with some local (non-Bayesian) optimization? Are you familiar with this paper, which does something along these lines? http://proceedings.mlr.press/v80/mcleod18a/mcleod18a.pdf

**Reply by sambitmishra98** [reply]
Thanks, I will check that out.
